packages/editor/src/components/media-categories/index.js

```diff
@@ -0,0 +1,182 @@
+ *
+ * @typedef {Object} InserterMediaCategory
+ * @property {string}                                                 label                The label of the media category. It's used in the inserter media items list.
+ * @property {string}                                                 [searchLabel]        The label to append in the search placeholder. If not provided the lowercase `label` will be used.
+ * @property {string}                                                 name                 The name of the media category, that should be unique among all media cateogries.
+ * @property {('image'|'audio'|'video')}                              mediaType            The media type of the media category.
+ * @property {(InserterMediaRequest) => Promise<InserterMediaItem[]>} fetch                The function to fetch media items for the category.
+ * @property {(InserterMediaItem)=> string}                           [getReportUrl]       If the media category support reporting media items, this function should return
+ *                                                                                         the report url for the media item. It accepts the `InserterMediaItem` as an argument.
+ * @property {boolean}                                                [isExternalResource] If the media category is an external resource, this should be set to true.
+ *                                                                                         This is used to avoid making a request to the external resource when the user
+ *                                                                                         opens the inserter for the first time.
+ */
+
+const getExternalLink = ( url, text ) =>
+	`<a href="${ url }" target="_blank" rel="noreferrer noopener">${ text }</a>`;
+
+const getOpenverseLicense = ( license, licenseVersion, licenseUrl ) => {
+	let licenseName = license.trim();
+	// PDM has no abbreviation
+	if ( license !== 'pdm' ) {
+		licenseName = license.toUpperCase().replace( 'SAMPLING', 'Sampling' );
+	}
+	// If version is known, append version to the name.
+	// The license has to have a version to be valid. Only
+	// PDM (public domain mark) doesn't have a version.
+	if ( licenseVersion ) {
+		licenseName += ` ${ licenseVersion }`;
+	}
+	// For licenses other than public-domain marks, prepend 'CC' to the name.
+	if ( ! [ 'pdm', 'cc0' ].includes( license ) ) {
+		licenseName = `CC ${ licenseName }`;
+	}
+	if ( !! licenseUrl ) {
+		license = getExternalLink(
+			`${ licenseUrl }?ref=openverse`,
+			licenseName
+		);
+	}
+	return license;
+};
+
+const getOpenverseCaption = ( item ) => {
+	const {
+		title,
+		foreign_landing_url: foreignLandingUrl,
+		creator,
+		creator_url: creatorUrl,
+		license,
+		license_version: licenseVersion,
+		license_url: licenseUrl,
+	} = item;
+	/* translators: Openverse default media item title in the block inserter's media list. (ex. 'This work by {creator} is marked with CC0 1.0'). */
```

Comment:
This function should revolve around just two possible strings:

* `'This work by %1$s is marked with %2$s'`
* `'"%1$s" by %2$s is marked with %3$s'`

We can't be stitching things up like `"${ _title }" ${ creator }/ ...`